api/include/opentelemetry/nostd/type_traits.h

```diff
@@ -6,19 +6,9 @@
 #if defined(OPENTELEMETRY_STL_VERSION)
 #  if OPENTELEMETRY_STL_VERSION >= 2011
 #    include "opentelemetry/std/type_traits.h"
-#    define OPENTELEMETRY_HAVE_STD_TYPE_TRAITS
 #  endif
 #endif
 
-#if !defined(OPENTELEMETRY_HAVE_STD_TYPE_TRAITS)
-#  include <array>
```

Comment:
`#if !defined(OPENTELEMETRY_HAVE_STD_TYPE_TRAITS)` Can be removed, but these headers are still required.